packages/alea-frontend/pages/course-home/[courseId].tsx

```diff
@@ -309,6 +311,63 @@ function CourseScheduleSection({
           </Box>
         )}
 
+        {examDates.length > 0 && (
```

Comment:
This code looks good. 

But this component has become too long. Also, this is uncoupled from the rest of the parts of the component. Please refactor this out into a separate component.